### ISSUE TITLE
Build error, cannot find symbol OSEMailSubscriptionObserver [IONIC][ANDROID]

### ISSUE THREAD

**Issue by jsnkelvin** [OneSignal/OneSignal-Cordova-SDK#318]
Hello, just installed the newest one signal plugin, and cannot build at all due to this error. Any help? 

Version : 
```
"onesignal-cordova-plugin": "^2.3.0"
"@ionic-native/onesignal": "^4.5.3"
```

Error Log
```
BUILD FAILED in 3s
27 actionable tasks: 1 executed, 26 up-to-date
Error: /Users/jasonjaputra/Documents/Projects/whatever/platforms/android/gradlew: Command failed with exit code 1 Error output:
/Users/jasonjaputra/Documents/Projects/whatever/platforms/android/src/com/plugin/gcm/OneSignalPush.java:108: error: cannot find symbol
  private static OSEmailSubscriptionObserver emailSubscriptionObserver;
                 ^
  symbol:   class OSEmailSubscriptionObserver
  location: class OneSignalPush
/Users/jasonjaputra/Documents/Projects/whatever/platforms/android/src/com/plugin/gcm/OneSignalPush.java:215: error: cannot find symbol
        emailSubscriptionObserver = new OSEmailSubscriptionObserver() {
                                        ^
  symbol:   class OSEmailSubscriptionObserver
  location: class OneSignalPush
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors
```

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileDebugJavaWithJavac'.
```

Ionic info 
cli packages: (/usr/local/lib/node_modules)

    @ionic/cli-utils  : 1.19.1
    ionic (Ionic CLI) : 3.19.1

global packages:

    cordova (Cordova CLI) : 7.1.0 

local packages:

    @ionic/app-scripts : 3.1.8
    Cordova Platforms  : android 6.4.0
    Ionic Framework    : ionic-angular 3.9.2

System:

    Android SDK Tools : 26.0.2
    ios-deploy        : 1.9.2 
    ios-sim           : 5.1.0 
    Node              : v8.0.0
    npm               : 5.6.0 
    OS                : macOS High Sierra
    Xcode             : Xcode 9.2 Build version 9C40b

**Comment by kodeine** [comment]
i am having the same issue!

**Comment by Nightsd01** [comment]
Thank you for your excellent bug report, we will investigate this issue and will get back to you shortly. In the mean time I would suggest downgrading to the previous version so it doesn’t impact your development.

**Comment by spinninghamster** [comment]
Same problem here, only started today.

> --------------------------------------------------------------------------------
> COMPILE OUTPUT
> --------------------------------------------------------------------------------
> Configuration on demand is an incubating feature.
> :wrapper
> 
> BUILD SUCCESSFUL in 0s
> 1 actionable task: 1 executed
> Subproject Path: CordovaLib
> Configuration on demand is an incubating feature.
> The Task.leftShift(Closure) method has been deprecated and is scheduled to be removed in Gradle 5.0. Please use Task.doLast(Action) instead.
> 	at build_capdh4h8lnfzhvm748mwd7n4g.run(/build.gradle:138)
> The JavaCompile.setDependencyCacheDir() method has been deprecated and is scheduled to be removed in Gradle 4.0.
> Incremental java compilation is an incubating feature.
> The TaskInputs.source(Object) method has been deprecated and is scheduled to be removed in Gradle 4.0. Please use TaskInputs.file(Object).skipWhenEmpty() instead.
> Download https://maven.google.com/com/google/android/gms/play-services-maps/11.2.2/play-services-maps-11.2.2.pom
> Download https://maven.google.com/com/google/android/gms/play-services-maps/11.2.2/play-services-maps-11.2.2.aar
> :preBuild UP-TO-DATE
> :preDebugBuild UP-TO-DATE
> :checkDebugManifest
> :preReleaseBuild UP-TO-DATE
> :CordovaLib:preBuild UP-TO-DATE
> :CordovaLib:preDebugBuild UP-TO-DATE
> :CordovaLib:checkDebugManifest
> :CordovaLib:prepareDebugDependencies
> :CordovaLib:compileDebugAidl
> :CordovaLib:compileDebugNdk UP-TO-DATE
> :CordovaLib:compileLint
> :CordovaLib:copyDebugLint UP-TO-DATE
> :CordovaLib:mergeDebugShaders
> :CordovaLib:compileDebugShaders
> :CordovaLib:generateDebugAssets
> :CordovaLib:mergeDebugAssets
> :CordovaLib:mergeDebugProguardFiles
> :CordovaLib:packageDebugRenderscript UP-TO-DATE
> :CordovaLib:compileDebugRenderscript
> :CordovaLib:generateDebugResValues
> :CordovaLib:generateDebugResources
> :CordovaLib:packageDebugResources
> :CordovaLib:processDebugManifest
> :CordovaLib:generateDebugBuildConfig
> :CordovaLib:processDebugResources
> :CordovaLib:generateDebugSources
> :CordovaLib:incrementalDebugJavaCompilationSafeguard
> :CordovaLib:compileDebugJavaWithJavac
> :CordovaLib:compileDebugJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
> Note: Some input files use or override a deprecated API.
> Note: Recompile with -Xlint:deprecation for details.
> :CordovaLib:processDebugJavaRes UP-TO-DATE
> :CordovaLib:transformResourcesWithMergeJavaResForDebug
> :CordovaLib:transformClassesAndResourcesWithSyncLibJarsForDebug
> :CordovaLib:mergeDebugJniLibFolders
> :CordovaLib:transformNative_libsWithMergeJniLibsForDebug
> :CordovaLib:transformNative_libsWithSyncJniLibsForDebug
> :CordovaLib:bundleDebug
> :CordovaLib:preReleaseBuild UP-TO-DATE
> :CordovaLib:checkReleaseManifest
> :CordovaLib:prepareReleaseDependencies
> :CordovaLib:compileReleaseAidl
> :CordovaLib:compileReleaseNdk UP-TO-DATE
> :CordovaLib:copyReleaseLint UP-TO-DATE
> :CordovaLib:mergeReleaseShaders
> :CordovaLib:compileReleaseShaders
> :CordovaLib:generateReleaseAssets
> :CordovaLib:mergeReleaseAssets
> :CordovaLib:mergeReleaseProguardFiles
> :CordovaLib:packageReleaseRenderscript UP-TO-DATE
> :CordovaLib:compileReleaseRenderscript
> :CordovaLib:generateReleaseResValues
> :CordovaLib:generateReleaseResources
> :CordovaLib:packageReleaseResources
> :CordovaLib:processReleaseManifest
> :CordovaLib:generateReleaseBuildConfig
> :CordovaLib:processReleaseResources
> :CordovaLib:generateReleaseSources
> :CordovaLib:incrementalReleaseJavaCompilationSafeguard
> :CordovaLib:compileReleaseJavaWithJavac
> :CordovaLib:compileReleaseJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
> Note: Some input files use or override a deprecated API.
> Note: Recompile with -Xlint:deprecation for details.
> :CordovaLib:processReleaseJavaRes UP-TO-DATE
> :CordovaLib:transformResourcesWithMergeJavaResForRelease
> :CordovaLib:transformClassesAndResourcesWithSyncLibJarsForRelease
> :CordovaLib:mergeReleaseJniLibFolders
> :CordovaLib:transformNative_libsWithMergeJniLibsForRelease
> :CordovaLib:transformNative_libsWithSyncJniLibsForRelease
> :CordovaLib:bundleRelease
> :prepareComAndroidSupportAnimatedVectorDrawable2540Library
> :prepareComAndroidSupportAppcompatV72540Library
> :prepareComAndroidSupportCustomtabs2540Library
> :prepareComAndroidSupportSupportCompat2540Library
> :prepareComAndroidSupportSupportCoreUi2540Library
> :prepareComAndroidSupportSupportCoreUtils2540Library
> :prepareComAndroidSupportSupportFragment2540Library
> :prepareComAndroidSupportSupportMediaCompat2540Library
> :prepareComAndroidSupportSupportV42540Library
> :prepareComAndroidSupportSupportVectorDrawable2540Library
> :prepareComGoogleAndroidGmsPlayServicesAnalytics1122Library
> :prepareComGoogleAndroidGmsPlayServicesAnalyticsImpl1122Library
> :prepareComGoogleAndroidGmsPlayServicesBase1122Library
> :prepareComGoogleAndroidGmsPlayServicesBasement1122Library
> :prepareComGoogleAndroidGmsPlayServicesGcm1122Library
> :prepareComGoogleAndroidGmsPlayServicesIid1122Library
> :prepareComGoogleAndroidGmsPlayServicesLocation1122Library
> :prepareComGoogleAndroidGmsPlayServicesMaps1122Library
> :prepareComGoogleAndroidGmsPlayServicesTagmanagerV4Impl1122Library
> :prepareComGoogleAndroidGmsPlayServicesTasks1122Library
> :prepareComOnesignalOneSignal381Library
> :prepareOrgApacheCordovaCordovaLib623DebugLibrary
> :prepareDebugDependencies
> :compileDebugAidl
> :compileDebugRenderscript
> :generateDebugBuildConfig
> :generateDebugResValues
> :generateDebugResources
> :mergeDebugResources
> :processDebugManifest
> :processDebugResources
> :generateDebugSources
> :incrementalDebugJavaCompilationSafeguard
> :compileDebugJavaWithJavac
> :compileDebugJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
> /src/com/plugin/gcm/OneSignalPush.java:108: error: cannot find symbol
>   private static OSEmailSubscriptionObserver emailSubscriptionObserver;
>                  ^
>   symbol:   class OSEmailSubscriptionObserver
>   location: class OneSignalPush
> /src/com/plugin/gcm/OneSignalPush.java:215: error: cannot find symbol
>         emailSubscriptionObserver = new OSEmailSubscriptionObserver() {
>                                         ^
>   symbol:   class OSEmailSubscriptionObserver
>   location: class OneSignalPush
> Note: Some input files use or override a deprecated API.
> Note: Recompile with -Xlint:deprecation for details.
> Note: Some input files use unchecked or unsafe operations.
> Note: Recompile with -Xlint:unchecked for details.
> 2 errors
> :compileDebugJavaWithJavac FAILED
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':compileDebugJavaWithJavac'.
> > Compilation failed; see the compiler error output for details.
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
> 
> BUILD FAILED
> 
> Total time: 3.266 secs
> Error: /gradlew: Command failed with exit code 1 Error output:
> Note: Some input files use or override a deprecated API.
> Note: Recompile with -Xlint:deprecation for details.
> Note: Some input files use or override a deprecated API.
> Note: Recompile with -Xlint:deprecation for details.
> /src/com/plugin/gcm/OneSignalPush.java:108: error: cannot find symbol
>   private static OSEmailSubscriptionObserver emailSubscriptionObserver;
>                  ^
>   symbol:   class OSEmailSubscriptionObserver
>   location: class OneSignalPush
> /src/com/plugin/gcm/OneSignalPush.java:215: error: cannot find symbol
>         emailSubscriptionObserver = new OSEmailSubscriptionObserver() {
>                                         ^
>   symbol:   class OSEmailSubscriptionObserver
>   location: class OneSignalPush
> Note: Some input files use or override a deprecated API.
> Note: Recompile with -Xlint:deprecation for details.
> Note: Some input files use unchecked or unsafe operations.
> Note: Recompile with -Xlint:unchecked for details.
> 2 errors
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':compileDebugJavaWithJavac'.
> > Compilation failed; see the compiler error output for details.
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
>     at ChildProcess.whenDone (/cordova/node_modules/cordova-common/src/superspawn.js:169:23)
>     at emitTwo (events.js:106:13)
>     at ChildProcess.emit (events.js:191:7)
>     at maybeClose (internal/child_process.js:877:16)
>     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)

**Comment by nowrap** [comment]
Downgrading to version 2.2.5 worked so far:
`ionic cordova plugin remove onesignal-cordova-plugin`
`ionic cordova plugin add onesignal-cordova-plugin@2.2.5`

My ionic info:

> cli packages: (C:\Users\nowrap\AppData\Roaming\npm\node_modules)
> 
>     @ionic/cli-utils  : 1.19.1
>     ionic (Ionic CLI) : 3.19.1
> 
> global packages:
> 
>     cordova (Cordova CLI) : 7.1.0
> 
> local packages:
> 
>     @ionic/app-scripts : 3.1.8
>     Cordova Platforms  : android 6.4.0 ios 4.5.3
>     Ionic Framework    : ionic-angular 3.9.2
> 
> System:
> 
>     Node : v8.9.4
>     npm  : 5.7.0
>     OS   : Windows 10

**Comment by Nightsd01** [comment]
Thank you for reporting this issue, it is now fixed in version 2.3.1

**Comment by kamran377** [comment]
Thanks for the quick fix
